### PR TITLE
Bugfix FXIOS-5687 [v123] EnhancedTrackingProtectionMenuVC is not adjusting font size according to users preference

### PR DESCRIPTION
**Pull request by imsanchit**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5687)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/13109)

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Browser/EnhancedTrackingProtection/SlideoverPresentationController.swift [hunk 1]
```diff
@@ -29,12 +27,7 @@ class SlideOverPresentationController: UIPresentationController {
     }
 
     override var frameOfPresentedViewInContainerView: CGRect {
-        var menuHeight: CGFloat
-        if globalETPStatus {
-            menuHeight = SlideOverUXConstants.ETPMenuHeightForGlobalOn
-        } else {
-            menuHeight = SlideOverUXConstants.ETPMenuHeightForGlobalOff
-        }
+        let menuHeight: CGFloat = presentedView!.systemLayoutSizeFitting(CGSize(width: presentedView!.bounds.width, height: 0.0), withHorizontalFittingPriority: .required, verticalFittingPriority: .defaultLow).height
```
Comment (adudenamedruby) [inline Client/Frontend/Browser/EnhancedTrackingProtection/SlideoverPresentationController.swift:30]: We'll soon be enabling a swiftlint rule that doesn't allow force unwrapping, so can we saftely unwrap this?

**Review by adudenamedruby — CHANGES_REQUESTED** [review]
Heya. Thanks for this work. Would you mind adding some before/after screenshots for UI work.

also tagging @thatswinnie for accessibility stuff :D

**Comment by imsanchit** [comment]
hey @adudenamedruby, thanks for reviewing the PR, I am attaching below both videos and screenshots of before and after PR states. Please have a look and let me know if anything else is needed.

**Before PR**
before-pr

https://github.com/mozilla-mobile/firefox-ios/assets/15265718/1e0d5148-1556-4535-8edd-bd363c86d594

**After PR**
after-pr

https://github.com/mozilla-mobile/firefox-ios/assets/15265718/e49abc3d-e703-40c4-9815-a4a55b0cb043

File: Client/Frontend/Browser/EnhancedTrackingProtection/SlideoverPresentationController.swift (hunk 1 above)
Comment (imsanchit) [reply Client/Frontend/Browser/EnhancedTrackingProtection/SlideoverPresentationController.swift]: Added an if-let block to do safe unwrap.

**Comment by adudenamedruby** [comment]
Thanks for the changes.

👀  taking a loot today/tomorrow.

**Comment by adudenamedruby** [comment]
Heya. Things look good, but I ran it on my computer, and my personal device, and I see the following on iPhone. But on iPad, it looks fine. Would you mind taking a look and seeing what's going on?


https://github.com/mozilla-mobile/firefox-ios/assets/11182210/bca09d4a-64b2-4646-a8d7-3321e422ebb9

**Comment by imsanchit** [comment]
Hey, thanks @adudenamedruby for pointing this out, have added a fix for this, please have a look.

**Comment by adudenamedruby** [comment]
Amazing. Thanks so much for that. Looking over the thing, I noticed a weird behaivour - the view behind the popup becomes visible at the bottom -that would need to be address before we can merge this PR. 

https://github.com/mozilla-mobile/firefox-ios/assets/11182210/c52dc510-a3ab-4333-b9da-3f3a8cd25c62

**Comment by imsanchit** [comment]
hey thank you once again, have updated the PR with a fix, could you please check this issue again?

**Review by adudenamedruby — APPROVED** [review]
Ran it and it works well on both iPhone & iPad.

Great work and thanks so much for your contribution!!!!!

**Comment by adudenamedruby** [comment]
Looks like there's a Swiftlint error that needs to be fixed before pipeline will run.

Screenshot 2023-12-18 at 10 33 20 AM

**Pull request merged**